modules/geoid/src/parse-pgm.js

```diff
@@ -0,0 +1,171 @@
+// Original code attribution:
+// Copyright (c) Charles Karney (2009-2018) <charles@karney.com> and licensed
+// under the MIT/X11 License.
+
+import Geoid from './geoid';
+
+const ENDL = 10;
+const PIXEL_MAX = 65535;
+
+// eslint-disable-next-line
+export function parsePGM(data, options) {
+  const getline = _getLineGenerator(data);
+  let currentLine = getline.next();
+  if (currentLine.done || currentLine.value.line !== 'P5') {
+    throw new Error('Geoid model file: File not in PGM format');
+  }
+  let _offset = Number.MAX_VALUE;
+  let _scale = 0;
+  let _maxerror = -1;
+  let _rmserror = -1;
+  let _description = 'NONE';
+  let _datetime = 'UNKNOWN';
+  let _width = 0;
+  let _height = 0;
+  let _datastart = null;
+  let _swidth = null;
+  do {
+    currentLine = getline.next();
+    const s = currentLine.value.line;
+    if (!s.length) {
+      continue; // eslint-disable-line no-continue
+    }
+    if (s[0] === '#') {
+      const sArr = s.split(' ');
+      const commentId = sArr[0];
+      const key = sArr[1];
+      if (commentId !== '#' || !key) {
+        continue; // eslint-disable-line no-continue
+      }
+      const infoArr = sArr.length > 2 ? sArr.slice(2) : [];
+      if (key === 'Description') {
+        _description = infoArr.join(' ');
+      } else if (key === 'DateTime') {
+        _datetime = infoArr.join(' ');
+      } else if (key === 'Offset') {
+        // eslint-disable-next-line max-depth
+        if (!sArr[2]) {
+          throw new Error('Geoid model file: Error reading offset');
+        }
+        _offset = parseInt(sArr[2], 10);
+      } else if (key === 'Scale') {
+        // eslint-disable-next-line max-depth
+        if (!sArr[2]) {
+          throw new Error('Geoid model file: Error reading scale');
+        }
+        _scale = parseFloat(sArr[2]);
+      } else if (key === (options.cubic ? 'MaxCubicError' : 'MaxBilinearError')) {
+        // It's not an error if the error can't be read
+        // eslint-disable-next-line max-depth
+        if (isFinite(parseFloat(sArr[2]))) {
+          _maxerror = parseFloat(sArr[2]);
+        }
+      } else if (key === (options.cubic ? 'RMSCubicError' : 'RMSBilinearError')) {
+        // It's not an error if the error can't be read
+        // eslint-disable-next-line max-depth
+        if (isFinite(parseFloat(sArr[2]))) {
+          _rmserror = parseFloat(sArr[2]);
+        }
+      }
+    } else {
+      let sArr = s.split(' ');
+      sArr = sArr.filter(testString => testString !== '');
+      _width = parseInt(sArr[0], 10);
+      _height = parseInt(sArr[1], 10);
+      if (!(_width && _height)) {
+        throw new Error('Geoid model file: Error reading raster size');
+      }
+      break;
+    }
+  } while (!currentLine.done);
+
+  currentLine = getline.next();
+  const maxval = parseInt(currentLine.value.line, 10);
+  if (currentLine.done) {
+    throw new Error('Geoid model file: Error reading maxval');
```

Comment:
Ideally, I think we should split this parser into two function. One which parses a PGM generically, without knowing that it will be used for EGM / Geoid. 

Then another function that uses that result to initialize the Geoid.